integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/rule/InvalidRowPercentRuleImpl.java

```diff
@@ -0,0 +1,135 @@
+ * Percent of Invalid rows out of all rows less than invalid threshold defined by
+ * DQ_INVALID_ALLOWED_PERCENT_ATTRIBUTE
+ * 
+ * If no threshold attribute is defined, the default value is used which is 0% i.e. no invalid rows.
+ */
+public class InvalidRowPercentRuleImpl implements DataQualityRule {
+
+    private static final Logger log = LoggerFactory.getLogger(InvalidRowPercentRuleImpl.class);
+
+    private String name;
+    private String description;
+    private boolean status;
+
+    private long feedRowCount;
+    private long invalidRowCount;
+    private long invalidThreshold;
+    private double pctInvalid;
+
+    public InvalidRowPercentRuleImpl() {
+        this.name = "INVALID_ROW_PERCENT_RULE";
+        this.description = "Percent of (Invalid rows / Total rows) <= " +
+                           DataQualityConstants.DEFAULT_INVALID_ALLOWED_PERCENT_VALUE;
+        this.status = false;
+        this.pctInvalid = Double.MAX_VALUE;
+    }
+
+    @Override
+    public boolean loadAttributes(FlowAttributes flowAttr) {
+        log.info("Loading Attributes for rule: " + name);
+        try {
+            feedRowCount = flowAttr.getAttributeValueLong(DataQualityConstants.DQ_FEED_ROW_COUNT_ATTRIBUTE);
+            invalidRowCount = flowAttr.getAttributeValueLong(DataQualityConstants.DQ_INVALID_ROW_COUNT_ATTRIBUTE);
+            invalidThreshold = flowAttr.getAttributeValueLong(DataQualityConstants.DQ_INVALID_ALLOWED_PERCENT_ATTRIBUTE,
+                                                              DataQualityConstants.DEFAULT_INVALID_ALLOWED_PERCENT_VALUE);
+        } catch (MissingAttributeException e) {
+            log.error("Required attributes missing");
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getDescription() {
+        return description;
+    }
+
+    @Override
+    public boolean getStatus() {
+        return status;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public JSONObject getSummary() {
+        JSONObject jsonAttrs = new JSONObject();
+        jsonAttrs.put("Invalid Row Count", invalidRowCount);
+        jsonAttrs.put("Invalid Threshold", invalidThreshold);
+        jsonAttrs.put("Feed Row Count", feedRowCount);
+        jsonAttrs.put("Percent Invalid", pctInvalid);
+
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("DESCRIPTION", getDescription());
+        jsonObject.put("STATUS", status);
+        jsonObject.put("ATTRIBUTES", jsonAttrs);
+
+        return jsonObject;
+    }
+
+    @Override
+    public boolean evaluate() {
+        try {
+            // Execute rule
+            pctInvalid = (invalidRowCount / ((double) feedRowCount)) * 100;
+            status = (pctInvalid <= invalidThreshold);
+
+            if (!status) {
+                log.warn("Invalid Row Count = " + invalidRowCount
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2629)